geode-core/src/main/java/org/apache/geode/cache/execute/FunctionService.java

```diff
@@ -72,7 +72,7 @@ protected FunctionService(FunctionExecutionService functionExecutionService) {
    * @throws FunctionException if the region passed in is null
    * @since GemFire 6.0
    */
-  public static Execution onRegion(Region region) {
+  public static <I, O, A> Execution<I, O, A> onRegion(Region region) {
```

Comment:
I know it is normal to see single letter generic arguments in Java but I feel like these 3 are not very obvious. Perhaps using complete names rather than a single letter would help. Since this is public API it is also prudent to update the Javadocs to describe the generic arguments.

Reply:
@pivotal-jbarrett Yes, I was on the fence about that myself. Single characters are idiomatic but words are more descriptive. Still not sure which is best to be honest.

Reply:
https://google.github.io/styleguide/javaguide.html#s5.2.8-type-variable-names

Here is what our style guide says:
5.2.8 Type variable names

Each type variable is named in one of two styles:

A single capital letter, optionally followed by a single numeral (such as E, T, X, T2)
A name in the form used for classes (see Section 5.2.2, Class names), followed by the capital letter T (examples: RequestT, FooBarT).

Reply:
@pivotal-jbarrett updated the code to use the convention used in existing code of `Execution` interface etc. ie. `<IN, OUT, AGG>`.